Lunar Festival Fixes

Oroxzy opened the pull request:
- [x] Respawn Elders from sniffs. 
- [x] Add missing Objects (180755, 180756)
- [x] Fix Cluster Launcher rocket positions.
- [x] Add Moonglade Revelers and Objects to China Lunar event.

**Facts:**

- Calculated Rocket positions for launchers:
- https://docs.google.com/spreadsheets/d/1JJ-e3jW_4tIYoFX3_ORbnPS-evJxezaV-1Y8aABnBME/edit?usp=sharing

Oroxzy committed:
this is the way!

Oroxzy committed:
Alarte Ascendare

ratkosrb commented on src/game/Spells/SpellEffects.cpp:
```diff
@@ -5638,13 +5629,10 @@ void Spell::EffectActivateObject(SpellEffectIndex eff_idx)
             break;
         case GameObjectActions::AnimateCustom0:
             gameObjTarget->SendGameObjectCustomAnim(0);
-            break;
```
wtf are you doing here, now it will fall through

Oroxzy replied:
was just a test, i fixed it via spell_effect_mod

ratkosrb commented on src/game/Spells/SpellEffects.cpp:
```diff
@@ -697,15 +697,6 @@ void Spell::EffectDummy(SpellEffectIndex eff_idx)
-                    if (!m_caster->IsPlayer())
-                        return;
-
-                    m_caster->RemoveSpellCooldown(26373); // Remove cooldown from Lunar Invititation
```
why, this is needed to not set cooldown when u use it in wrong area

Oroxzy replied:
its wrong.. no one sniffed it before me, it just teleports you to the moonglade position

Oroxzy replied:
it was just a quick improvisation from myself, because nobody could tell me how it really works

Oroxzy committed:
next

Oroxzy committed:
Update 20220205202111_world.sql

Oroxzy committed:
Update 20220205202111_world.sql

Oroxzy committed:
Update 20220205202111_world.sql

Oroxzy committed:
Update 20220205202111_world.sql

Oroxzy committed:
Update 20220205202111_world.sql

Oroxzy committed:
Update 20220205202111_world.sql

Oroxzy committed:
Merge branch 'development' into lunar_festival_fixes

Oroxzy committed:
Update 20220205202111_world.sql

Oroxzy committed:
Update 20220205202111_world.sql

ratkosrb commented on sql/migrations/20220205202111_world.sql:
```diff
@@ -248,6 +248,43 @@ INSERT INTO `spell_effect_mod` (`Id`, `EffectMiscValue`, `Comment`) VALUES
+    (26409, 0, 25, 'Fix for Holiday - Lunar Teleport Cap: Darnassus'),
+    (26410, 0, 25, 'Fix for Holiday - Lunar Teleport Cap: Undercity'),
+    (26412, 0, 25, 'Fix for Holiday - Lunar Teleport Cap: Thunderbluff'),
+    (26414, 0, 25, 'Fix for Holiday - Lunar Teleport Cap: Orgrimmar');
```
Why is this needed? The spells seemed to already work? I remember i tested your last Lunar Festival PR and there were no problems.

Oroxzy replied:
it's not needed, i already removed these mods.

Oroxzy committed:
Update npcs_special.cpp

Oroxzy committed:
Update npcs_special.cpp

Oroxzy committed:
Update 20220205202111_world.sql

Reproduces the exact behavior-sequence from Reveler NPCs in Moonglade with the correct Spell

Oroxzy committed:
Update 20220205202111_world.sql

Oroxzy committed:
Merge branch 'development' into lunar_festival_fixes

ratkosrb commented on sql/migrations/20220205202111_world.sql:
```diff
@@ -278,6 +278,218 @@ INSERT INTO `conditions` (`condition_entry`, `type`, `value1`, `value2`, `value3
+
+INSERT INTO `spell_script_target` (`entry`, `type`, `targetEntry`, `conditionId`, `inverseEffectMask`, `build_min`, `build_max`) VALUES (26347, 0, 180771, 1111, 0, 5086, 5875);
+INSERT INTO `spell_script_target` (`entry`, `type`, `targetEntry`, `conditionId`, `inverseEffectMask`, `build_min`, `build_max`) VALUES (26347, 0, 180868, 1111, 0, 5086, 5875);
+INSERT INTO `spell_script_target` (`entry`, `type`, `targetEntry`, `conditionId`, `inverseEffectMask`, `build_min`, `build_max`) VALUES (26347, 0, 180850, 1111, 0, 5086, 5875);
```
Whats the point of the condition here given that the entry is part of this table? 180850 is not equal to 15882, so what does the condition do?

Oroxzy replied:
![Firework_explaination](https://user-images.githubusercontent.com/357606/164889036-26d7f8ab-6f9a-427e-b6e0-7af818a7de87.png)

ratkosrb committed:
Remove already existing npc text.
